File: AICodingWeb/Models/Supplier.cs
using System;
using System.Collections.Generic;

namespace AICodingWeb.Models
{
    public class Supplier
    {
        public int SupplierID { get; set; }
        public string SupplierName { get; set; }
        public int SupplierCategoryID { get; set; }
        public int PrimaryContactPersonID { get; set; }
        public int? AlternateContactPersonID { get; set; }
        public int DeliveryMethodID { get; set; }
        public int DeliveryCityID { get; set; }
        public int PostalCityID { get; set; }
        public string SupplierReference { get; set; }
        public string BankAccountName { get; set; }
        public string BankAccountBranch { get; set; }
        public string BankAccountCode { get; set; }
        public string BankAccountNumber { get; set; }
        public string BankInternationalCode { get; set; }
        public string PaymentDays { get; set; }
        public string InternalComments { get; set; }
        public string PhoneNumber { get; set; }
        public string FaxNumber { get; set; }
        public string WebsiteURL { get; set; }
        public string DeliveryAddressLine1 { get; set; }
        public string DeliveryAddressLine2 { get; set; }
        public string DeliveryPostalCode { get; set; }
        public string DeliveryLocation { get; set; }
        public string PostalAddressLine1 { get; set; }
        public string PostalAddressLine2 { get; set; }
        public string PostalPostalCode { get; set; }
        public int LastEditedBy { get; set; }
        public DateTime LastEditedWhen { get; set; }

        public virtual SupplierCategory SupplierCategory { get; set; }
        public virtual Person PrimaryContactPerson { get; set; }
        public virtual Person AlternateContactPerson { get; set; }
        public virtual DeliveryMethod DeliveryMethod { get; set; }
        public virtual City DeliveryCity { get; set; }
        public virtual City PostalCity { get; set; }
        public virtual ICollection<PurchaseOrder> PurchaseOrders { get; set; }
        public virtual ICollection<StockItemTransaction> StockItemTransactions { get; set; }
        public virtual ICollection<SupplierTransaction> SupplierTransactions { get; set; }
        public virtual ICollection<StockItem> StockItems { get; set; }
    }
} 

File: AICodingWeb/aicoding_web/app/models/supplier.py
from datetime import datetime
from app import db
from app.models.base import BaseModel
from app.models.custom_types import Geography
from sqlalchemy.orm import deferred

class Supplier(db.Model, BaseModel):
    """Supplier model"""
    
    __tablename__ = 'Suppliers'
    __table_args__ = {'schema': 'Purchasing'}
    
    SupplierID = db.Column(db.Integer, primary_key=True)
    SupplierName = db.Column(db.String(100), nullable=False)
    SupplierCategoryID = db.Column(db.Integer, db.ForeignKey('Purchasing.SupplierCategories.SupplierCategoryID'), nullable=False)
    PrimaryContactPersonID = db.Column(db.Integer, db.ForeignKey('Application.People.PersonID'), nullable=False)
    AlternateContactPersonID = db.Column(db.Integer, db.ForeignKey('Application.People.PersonID'))
    DeliveryMethodID = db.Column(db.Integer, db.ForeignKey('Application.DeliveryMethods.DeliveryMethodID'))
    DeliveryCityID = db.Column(db.Integer, db.ForeignKey('Application.Cities.CityID'), nullable=False)
    PostalCityID = db.Column(db.Integer, db.ForeignKey('Application.Cities.CityID'), nullable=False)
    SupplierReference = db.Column(db.String(20))
    PaymentDays = db.Column(db.Integer, nullable=False)
    PostalCode = db.Column(db.String(10), nullable=False)
    PhoneNumber = db.Column(db.String(20), nullable=False)
    FaxNumber = db.Column(db.String(20), nullable=False)
    WebsiteURL = db.Column(db.String(256), nullable=False)
    DeliveryAddressLine1 = db.Column(db.String(60), nullable=False)
    DeliveryAddressLine2 = db.Column(db.String(60))
    DeliveryPostalCode = db.Column(db.String(10), nullable=False)
    DeliveryLocation = deferred(db.Column(db.String))  # Deferred loading for geography column
    PostalAddressLine1 = db.Column(db.String(60), nullable=False)
    PostalAddressLine2 = db.Column(db.String(60))
    PostalPostalCode = db.Column(db.String(10), nullable=False)
    LastEditedBy = db.Column(db.Integer, db.ForeignKey('Application.People.PersonID'), nullable=False)
    ValidFrom = db.Column(db.DateTime, nullable=False)
    ValidTo = db.Column(db.DateTime, nullable=False)
    
    def __repr__(self):
        return f'<Supplier {self.SupplierID}>' 